cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/tests.cpp

```diff
@@ -654,6 +654,37 @@ void test26(bool cond)
 	if (ptr[-1] == 0) { return; } // GOOD: accesses buffer[1]
 }
 
+void test27(){
+	char *src = "";
+	char *dest = "abcdefgh";
+	const int size = 100;
+	int ind = 100;
+	char buffer[size];
+
+	strncpy(dest, src, strlen(dest)); // GOOD, strncpy will not read past null terminator of source
+		
+	if(ind < size){
+		buffer[ind] = 0; // GOOD: out of bounds, but inaccessible code
```

Comment:
I don't get a result here with or without the `reachable` change.  I *do* get a difference if I change it to:
```suggestion
		buffer[100] = 0; // GOOD: out of bounds, but inaccessible code
```
(I'm not sure why this matters; it looks like trivial data flow to me)

This change actually reveals a similar issue in the `cpp/static-buffer-overflow` query, which can be fixed by adding `reachable(access)` to `staticBufferBase`.